File: src/main/java/com/moovie/backend/domain/usecase/activities/GetUserActivities.java
package com.moovie.backend.domain.usecase.activities;

import com.moovie.backend.domain.model.UserActivity;
import com.moovie.backend.domain.repository.UserActivitiesRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GetUserActivities {

    private final UserActivitiesRepository userActivitiesRepository;

    public GetUserActivities(UserActivitiesRepository userActivitiesRepository) {
        this.userActivitiesRepository = userActivitiesRepository;
    }

    public List<UserActivity> invoke(String userId) {
        return userActivitiesRepository.getUserActivities(userId);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/activities/GetUserActivities.kt
package org.mobyle.domain.usecase.activities

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.model.UserActivity
import org.mobyle.domain.repository.UserActivitiesRepository

class GetUserActivities(private val repository: UserActivitiesRepository) {
    operator fun invoke(userId: String): List<UserActivity> = runBlocking {
        repository.getUserActivities(userId)
    }
}
